### PR TITLE
Fix for osx

### PR DESCRIPTION
**Pull request by marimeireles**
Hi @rapgenic, I had to make these small changes to be able to build it locally for osx.

### CODE REVIEW

**Comment by marimeireles** [comment]
Should we release a new version with your changes so I can test it on conda-forge?

**Comment by marimeireles** [comment]
> Should we release a new version with your changes so I can test it on conda-forge?

I just created patches for it, so no need anymore for now.

File: CMakeLists.txt [hunk 1]
```diff
@@ -117,12 +117,13 @@ if (GLFW3_OSMESA_BACKEND)
   pkg_check_modules(osmesa REQUIRED IMPORTED_TARGET osmesa)
   target_link_libraries(${EXECUTABLE_NAME} PkgConfig::osmesa)
   set(NOTEBOOK_TOOLKIT_CPU TRUE)
+  target_link_libraries(${EXECUTABLE_NAME} glfw)
 else()
   find_package(OpenGL REQUIRED)
-  target_link_libraries(${EXECUTABLE_NAME} OpenGL::OpenGL)
+  find_package(glfw3)
+  target_link_libraries(${EXECUTABLE_NAME} glfw OpenGL::GL)
```
Comment (rapgenic) [inline CMakeLists.txt]: Thanks for catching the OpenGL target name!

Regarding the glfw target see the previous comment

File: CMakeLists.txt [hunk 2]
```diff
@@ -117,12 +117,13 @@ if (GLFW3_OSMESA_BACKEND)
   pkg_check_modules(osmesa REQUIRED IMPORTED_TARGET osmesa)
   target_link_libraries(${EXECUTABLE_NAME} PkgConfig::osmesa)
   set(NOTEBOOK_TOOLKIT_CPU TRUE)
+  target_link_libraries(${EXECUTABLE_NAME} glfw)
 else()
   find_package(OpenGL REQUIRED)
-  target_link_libraries(${EXECUTABLE_NAME} OpenGL::OpenGL)
+  find_package(glfw3)
```
Comment (rapgenic) [inline CMakeLists.txt:122]: The fact that you needed to add a call to find package is quite strange, this should have already been done by `find_or_fetch` before. I might have a clue, but to be certain, what commands did you invoke cmake with?

File: CMakeLists.txt [hunk 3]
```diff
@@ -117,12 +117,13 @@ if (GLFW3_OSMESA_BACKEND)
...
+  target_link_libraries(${EXECUTABLE_NAME} glfw OpenGL::GL)
 endif()
 
-target_link_libraries(${EXECUTABLE_NAME} glfw)
```
Comment (rapgenic) [inline CMakeLists.txt:125]: Was there really the need to remove this line and put it again in both if branches before it? GLFW is a dependency in any case, I think it's correct down here

**Review by rapgenic — COMMENTED** [review]
Hi @marimeireles, thanks again for providing feedback on mac os, I've left a few comments on the code

**Comment by rapgenic** [comment]
> Should we release a new version with your changes so I can test it on conda-forge?

Will release when this is solved!

File: CMakeLists.txt (hunk 2 above)
Comment (marimeireles) [reply CMakeLists.txt:122]: Just `cmake -D CMAKE_INSTALL_PREFIX=$CONDA_PREFIX ..` which are the basic commands do run cmake inside a conda env.

File: CMakeLists.txt (hunk 3 above)
Comment (marimeireles) [reply CMakeLists.txt:125]: I thought it might be a problem for osmesa, but if you think it isn't, I fixed it :)

**Comment by rapgenic** [comment]
I'm sorry, but I still believe that the way I wrote it, at least in theory, is right, whereas your modifications do not really make sense to me:

GLFW is always a dependency, so the 
```cmake
target_link_libraries(${EXECUTABLE_NAME} glfw)
```

line should be outside the if that handles the opengl/osmesa choice.

OpenGL/Osmesa are mutually exclusive, so one branch of the if should only link OpenGL where the other should only link osmesa (as it was before):

```cmake
# This calls "find_package(glfw3)", no need to call it again
find_or_fetch_package(glfw3 ...)

# This if handles the Opengl/Osmesa choice which is chosen independently from GLFW
if (GLFW3_OSMESA_BACKEND)
  pkg_check_modules(osmesa REQUIRED IMPORTED_TARGET osmesa)
  target_link_libraries(${EXECUTABLE_NAME} PkgConfig::osmesa)
  set(NOTEBOOK_TOOLKIT_CPU TRUE)
else()
  find_package(OpenGL REQUIRED)
  target_link_libraries(${EXECUTABLE_NAME} OpenGL::OpenGL)
endif()

# The GLFW linking is always done here because it's independent from the previous choice
target_link_libraries(${EXECUTABLE_NAME} glfw)
```

You put a `find_package(glfw3)` inside that if, which in my opinion is wrong, because glfw has already been found or downloaded by `find_or_fetch` call before.

As of now, then, I cannot merge it, at least until I've understood what's the problem with my original code...

I've fixed the `OpenGL::GL` target, which I'd mistakenly wrote `OpenGL::OpenGL`, so if you don't mind, could you please try with the new version in master, and if it still won't work, share with me the cmake logfiles?

**Comment by marimeireles** [comment]
Right, I'll test it on macos and paste the problem here. I cannot do it now, it will take a while.
Thanks for the review.

**Comment by rapgenic** [comment]
I believe this is no longer necessary, thanks @marimeireles!

**Pull request closed without merging**